vm.js

```diff
@@ -1362,11 +1362,16 @@ Object.subclass('Squeak.Object',
                         this.float = 0.0;
                     } else
                         this.words = new Uint32Array(indexableSize);
-        } else // Bytes
+        } else { // Bytes
+            if (this.format >= 12) {
+                this.compiled = false;
+                this.ic = new Array();
+            }
             if (indexableSize > 0) {
                 // this.format |= -indexableSize & 3;       //deferred to writeTo()
                 this.bytes = new Uint8Array(indexableSize); //Methods require further init of pointers
             }
+        }
 
```

Comment:
Please use the same order of adding properties as in `installFromImage()`. Otherwise these methods get different hidden classes.